File: src/protocol/TemperatureDemandBuilder.cpp
#include "TemperatureDemandBuilder.h"
#include "TemperatureDemandCommand.h"

using namespace std;
using namespace communication;

TemperatureDemandBuilder::TemperatureDemandBuilder()
{}

TemperatureDemandBuilder::~TemperatureDemandBuilder()
{}

unique_ptr<Command> TemperatureDemandBuilder::create(const vector<uint8_t> &commandInBytes)
{
    auto command = make_unique<TemperatureDemandCommand>();

    return move(command);
}

File: mobile_client/app/src/main/java/protocol/TemperatureDemandBuilder.java
package protocol;

import java.io.Closeable;
import java.util.ArrayList;

public final class TemperatureDemandBuilder implements CommandBuilder, Closeable
	{
		public TemperatureDemandBuilder()
		{
		}
		public final void close()
		{
		}

		@Override
		public Command create(ArrayList<Integer> commandInBytes)
		{
			Command command  = new TemperatureDemandCommand();
			return command;
		}
	}